gpib_server.py

```diff
@@ -194,6 +195,20 @@ def query(self, c, data):
         ans = instr.read_raw()
         return str(ans).strip()
 
+    @setting(7, n_bytes='w', returns='s')
+    def read_raw(self, c, n_bytes=None):
+        """Read from the GPIB bus.  No .strip()
+
+        If specified, reads only the given number of bytes.
+        Otherwise, reads until the device stops sending.
+        """
+        instr = self.getDevice(c)
+        if n_bytes is None:
+            ans = instr.read_raw()
+        else:
+            ans = instr.read_raw(n_bytes)
+        return str(ans)
```

Comment:
As per comment above, I'd suggest saying `bytes(ans)` here. In python 2 `bytes` is actually equal to `str`, but it's still a helpful hint IMO that this can be arbitrary bytes, not just a text string. This distinction will become more important in python3, of course.